accepted/2021/install-location-per-architecture.md

```diff
@@ -138,34 +141,56 @@ any specific knowledge of where to find the install location, it would just
 invoke the right apphost.
 
 On all platforms the apphost will have a default location hardcoded
-and so if the installer puts the layout into that location it doesn't have to
+and so if the installer puts the product into that location it doesn't have to
 register that installation. But it would be cleaner if it did so anyway.
-Currently we always register the installation on Windows, but we don't seem
-to do so on Linux and macOS.
+Currently we always register the installation on Windows, but we don't do so
+on Linux and macOS.
 
-*Note: If installers write the file, they should also read it and validate
-that they don't overwrite it with different information. It's unclear yet
-if this is necessary and in which situations.*
+#### Windows installer behavior
 
-#### Support for downlevel apphost
+Pre-.NET 6 Windows installer writes registry keys for all architectures
+(even though it only installs one of them). Ideally the installer would
+only write the key for the architecture it's actually installing.
 
-In order to support running downlevel apphost apps on multi-arch non-Windows
-architectures (currently only applies to macOS), the installer of .NET 6 and above
-should write the configuration file in a specific shape, for example (on macOS arm64):
+#### Linux/macOS installer behavior
 
-```console
-/usr/local/share/dotnet/x64
-arm64=/usr/local/share/dotnet
-x64=/usr/local/share/dotnet/x64
-```
+Pre-.NET 6 Linux/macOS installer doesn't write the `install_location` file.
 
-With the above proposed changes this would guarantee that all apphosts
-would work if the right frameworks are installed.
+Starting with .NET 6, installers should write architecture specific file
+with the product location, in the format `install_location_<arch>`. So for example
+`install_location_x64` (all lower case). The content of the file should be
+a single line with the path to the product.
 
-* 3.1 and 5 apphost would only read the first line, which points to x64 install.
-Since 3.1 and 5 are only supported on x64 on macOS, that is the right location.
-* arm64 6 apphost would react to the `arm64` location
-* x64 6 apphost would react to the `x64` location
+*Note: If installers can write the file, they should also read it and validate
+that they don't overwrite it with different information.*
+
+#### Support for downlevel apphost
+
+In order to support running downlevel apphost apps on multi-arch non-Windows
+architectures (currently only applies to macOS), the installer for
+the architecture which .NET supports in versions lower than .NET 6
+should also write the product location to the arch-less location file
+`install_location`. This is needed for downlevel applications to correctly
+find the product.
```

Comment:
I'm assuming we should only do this if the user hasn't already set this path.  See below discussion of product/user state.

Reply:
Probably - but as mentioned that should be very rare. But I agree that probably should not overwrite it.

Reply:
Current thinking is to overwrite

Reply:
Added a note that installers will overwrite.